scripts/run_performance_job.py

```diff
@@ -1010,9 +1010,10 @@ def publish_dotnet_app_to_payload(payload_dir_name: str, csproj_path: str, self_
 
             # TODO: See if these commands are needed for linux as they were being called before but were failing.
             if args.os_group == "windows" or args.os_group == "osx":
-                RunCommand([*(agent_python.split(" ")), "-m", "pip", "install", "--user", "--upgrade", "pip"]).run()
-                RunCommand([*(agent_python.split(" ")), "-m", "pip", "install", "--user", "urllib3==1.26.19"]).run()
-                RunCommand([*(agent_python.split(" ")), "-m", "pip", "install", "--user", "requests"]).run()
+                break_system_packages = ["--break-system-packages"] if args.os_group == "osx" else []
```

Comment:
To double check, the new `--break-system-packages` will hopefully allow these calls to succeed on OSX because without it we get an error about the environment being externally managed?

Reply:
Is there a reason why we don't create a python environment (`venv`) and instead we use the global system python? cc: @LoopedBard3

Reply:
Thanks Parker. Looking at the log, this seems to be coming from the helix machine but the commands above are run on the macOS build image where I don't see `.venv` being activated https://dev.azure.com/dnceng/internal/_build/results?buildId=2862740&view=logs&j=007c6ee1-d4e8-565e-80d2-5b93cbca4c6a&t=e8845911-7767-5da3-8d86-0ee0806c6d02

Reply:
Ah, I see. I am not aware of any reason we are not activating the .venv there other than us not having added that to the flow. @DrewScoggins anything to add to that?